modules/improvedigitalBidAdapter.js

```diff
@@ -0,0 +1,415 @@
+
+            for (var syncCounter = 0; syncCounter < syncArray.length; syncCounter++) {
+              syncString += (syncString === "")? "document.writeln(\"" : "";
+              var syncInd = syncArray[syncCounter];
+              syncInd = syncInd.replace(/\//g, '\\\/');
+              syncString += "<img src=\\\"" + syncInd + "\\\"\/>";
+            }
+            syncString += (syncString === "")? "" : "\")";
+
+            var nurl = "";
+            if (bidObject.nurl && bidObject.nurl.length > 0) {
+              nurl = "<img src=\"" + bidObject.nurl + "\" width=\"0\" height=\"0\" style=\"display:none\">";
+            }
+            bid.ad = nurl + "<script>" + bidObject.adm + syncString + "</script>";
+            bid.bidderCode = IMPROVE_DIGITAL_BIDDER_CODE;
+            bid.cpm = parseFloat(bidObject.price);
+            bid.width = bidObject.w;
+            bid.height = bidObject.h;
+
+            bidmanager.addBidResponse(bidRequest.placementCode, bid);
+          } else {
+            if (bidObject.errorCode && bidObject.errorCode !== 0) {
+              bidRequest.status = CONSTANTS.STATUS.NO_BID;
+            } else {
+              continue;
+            }
+          }
+        }
+      }
+    }
+  };
+
+  baseAdapter.callBids = function(params){
+    // params will contain an array
+    var bidRequests = params.bids || [];
+    var loc = utils.getTopWindowLocation();
+    var requestParameters = {
+      singleRequestMode: false,
+      httpRequestType: this.idClient.CONSTANTS.HTTP_REQUEST_TYPE.GET,
+      callback: CALLBACK_FUNCTION,
+      secure: (loc.protocol === 'https:')? 1 : 0,
+      libVersion: this.LIB_VERSION
+    };
+
+    var normalizedBids = [];
+    for (var bidCounter = 0; bidCounter < bidRequests.length; bidCounter++) {
+      var normalizedBidRequest = this.getNormalizedBidRequest(bidRequests[bidCounter]);
+      normalizedBids.push(normalizedBidRequest);
+      if(bidRequests[bidCounter].params && bidRequests[bidCounter].params.singleRequest) {
+        requestParameters.singleRequestMode = true;
+      }
+    }
+
+    var request = this.idClient.createRequest(
+      normalizedBids,   // requestObject
+      requestParameters
+    );
+
+    if (request.errors && request.errors.length > 0) {
+      console.log("ID WARNING 0x01");
+    }
+
+    if(request && request.requests && request.requests[0]) {
+      for (var requestCounter = 0; requestCounter < request.requests.length; requestCounter++) {
+        if (request.requests[requestCounter].url) {
+          adloader.loadScript(request.requests[requestCounter].url, null);
+        }
+      }
+    }
+  }
+
+  // Export the callBids function, so that prebid.js can execute this function
+  // when the page asks to send out bid requests.
+  return {
+    LIB_VERSION: LIB_VERSION,
+    idClient: baseAdapter.idClient,
+    getNormalizedBidRequest: baseAdapter.getNormalizedBidRequest,
+    callBids: baseAdapter.callBids
+  };
+};
```

Comment:
this will need change to (see https://github.com/prebid/Prebid.js/pull/1459)
```
return Object.assign(this, {
    LIB_VERSION: LIB_VERSION,
    idClient: baseAdapter.idClient,
    getNormalizedBidRequest: baseAdapter.getNormalizedBidRequest,
    callBids: baseAdapter.callBids
  });
```